Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageTableViewCell.swift

```diff
@@ -0,0 +1,72 @@
+//
+
+import StreamChat
+import UIKit
+
+/// The cell that displays the message content of a dynamic type and layout.
+/// Once the cell is set up it is expected to be dequeued for messages with
+/// the same content and layout the cell has already been configured with.
+public typealias ChatMessageTableViewCell = _ChatMessageTableViewCell<NoExtraData>
+
+/// The cell that displays the message content of a dynamic type and layout.
+/// Once the cell is set up it is expected to be dequeued for messages with
+/// the same content and layout the cell has already been configured with.
+public final class _ChatMessageTableViewCell<ExtraData: ExtraDataTypes>: _TableViewCell {
+    public static var reuseId: String { "\(self)" }
+    
+    /// The message content view the cell is showing.
+    public private(set) var messageContentView: _ChatMessageContentView<ExtraData>?
+    
+    override public func setUp() {
+        super.setUp()
+        
+        selectionStyle = .none
+    }
+    
+    override public func setUpAppearance() {
+        super.setUpAppearance()
+        
+        backgroundColor = .clear
+        backgroundView = nil
+    }
+
+    override public func prepareForReuse() {
+        super.prepareForReuse()
+
+        messageContentView?.prepareForReuse()
+    }
+
+    /// Creates a message content view
+    /// - Parameters:
+    ///   - contentViewClass: The type of message content view.
+    ///   - attachmentViewInjectorType: The type of attachment injector.
+    ///   - options: The layout options describing the message content view layout.
+    public func setMessageContentIfNeeded(
+        contentViewClass: _ChatMessageContentView<ExtraData>.Type,
+        attachmentViewInjectorType: _AttachmentViewInjector<ExtraData>.Type?,
+        options: ChatMessageLayoutOptions
+    ) {
+        guard messageContentView == nil else {
+            log.assert(type(of: messageContentView!) == contentViewClass, """
+            Attempt to setup different content class: ("\(contentViewClass)")
+            """)
+            return
+        }
+
+        messageContentView = contentViewClass.init().withoutAutoresizingMaskConstraints
+        // We add the content view to the view hierarchy before invoking `setUpLayoutIfNeeded`
+        // (where the subviews are instantiated and configured) to use `components` and `appearance`
+        // taken from the responder chain.
+        contentView.addSubview(messageContentView!)
+        messageContentView?.pin(anchors: [.leading, .top, .trailing, .bottom], to: contentView)
+        messageContentView!.setUpLayoutIfNeeded(options: options, attachmentViewInjectorType: attachmentViewInjectorType)
+        
+        if options.contains(.continuousBubble) && !options.contains(.threadInfo) {
+            messageContentView?.mainContainer.layoutMargins.bottom = 2
+        } else {
+            messageContentView?.mainContainer.layoutMargins.bottom = 8
```

Comment:
nit: As mention on the call this shouldn't depend on the option and it should be ideally easily set from the outside.

Reply:
✅ Fixed in 7f1803a924547bbead2b310e7e47e13ded16f528